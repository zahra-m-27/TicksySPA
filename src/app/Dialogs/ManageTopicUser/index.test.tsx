import ReactDOM from 'react-dom';
import React, {useRef} from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ManageTopicUserDialog from './index';
import user from '@testing-library/user-event';

const Sample: React.FC = ({}) => {
  const ref = useRef(() => {});
  return <ManageTopicUserDialog topicId={1} onDismissRef={ref} />;
};

const sample = <Sample />;

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render header', () => {
  const {getByText} = render(sample);
  getByText(/افزودن کاربر/i);
});

it('render email label', () => {
  const {getByText} = render(sample);
  getByText(/ایمیل/i);
});

it('render role label', () => {
  const {getByText} = render(sample);
  getByText(/عنوان نقش/i);
});

it('render buttons', () => {
  const {getByText} = render(sample);
  getByText(/کاربر جدید/i);
  getByText(/ثبت/i);
});

it('open new dialog', async () => {
  const {getByText, findByTestId} = render(sample);
  const newRoleButton = getByText(/کاربر جدید/i);
  user.click(newRoleButton);
  const dialogContainer = await findByTestId('dialog-container');
  user.click(dialogContainer);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
