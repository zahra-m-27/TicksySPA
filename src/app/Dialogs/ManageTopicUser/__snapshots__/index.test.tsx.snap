// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <span>
      افزودن کاربر
    </span>
    <img
      src="add_user_menu.svg"
    />
  </div>
  <div
    className="content"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    <img
      className="avatar"
      src="add_user.svg"
    />
    <div
      className="container"
    >
      <label
        className="label"
      >
        ایمیل
      </label>
      <div
        className="input_container"
        data-testid="input-container"
      >
        <img
          alt="icon"
          className="icon email_icon"
          src="search.png"
        />
        <input
          className="input_class"
          data-testid="email"
          dir="rtl"
          disabled={false}
          onChange={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
        />
      </div>
      <div
        className="auto_complete_items"
        data-active={false}
      />
    </div>
    <div
      className="container"
      data-open={false}
    >
      <p
        className="label"
      >
        عنوان نقش
      </p>
      <div
        className="content"
        data-testid="dropdown-container"
      >
        <img
          alt="arrow down"
          className="arrow_down"
          src="arrow_down.svg"
        />
        <p
          className="selected_item_label"
        >
          انتخاب كنيد
        </p>
      </div>
      <div
        className="items"
      />
    </div>
    <div
      className="buttons"
    >
      <div
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      >
        <span
          className="label"
        >
          کاربر جدید
        </span>
      </div>
      <div
        className="button"
        data-testid="submit_user"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#1354ac",
          }
        }
      >
        <span
          className="label"
        >
          ثبت
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning loading"
    style={
      Object {
        "display": undefined,
      }
    }
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;
